### PR TITLE
Added secondary email to all the places

### LINKED ISSUE #568: Add secondary email address, add checkboxes to denote email as out of service.
Steps to fix:
- Add database fields for secondary email
- Add database fields for declaring an email address out-of-use (and re-enabling) the primary and secondary emails. Default : mail address is in-use
-  and add small button in the form for declaring the primary and secondary email out-of-use.
-  Also update form logic to use said checkboxes
-  Update member log  (description below)

Do not forget to update the 'MemberLog' funcationality as well, which is where we store the historic data about what data we knew about a member in the past. Even though that table is currently not exposed to the end-user at all, we dó still use it as an audit trail.

### PR DESCRIPTION
**Pull request by RemcoSwenker**
Closes #568

### CODE REVIEW

File: datamining/views.py [hunk 1]
```diff
@@ -66,8 +66,8 @@ def show_members_by_group(request):
     today = datetime.date.today().strftime("%Y-%m-%d")
 
     for member in found_members:
-        if len(member.email) > 0:
-            member_mails.append(member.email)
+        if len(member.get_email()) > 0:
```
Comment (nanderv) [inline datamining/views.py:69]: Doesn't member.get_email just return a singular email address?

File: members/models/member.py [hunk 2]
```diff
@@ -61,6 +61,14 @@ def get_current_membership_period(self, current_date: datetime.date = None):
     def get_periods(self):
         return self.membershipperiod_set.all().order_by('start_date')
 
+    def get_email(self):
+        if self.primary_email_in_use:
+            return self.primary_email
+        elif self.secondary_email_in_use:
+            return self.secondary_email
+        else:
+            return self.primary_email
```
Comment (nanderv) [inline members/models/member.py:70]: Why send emails to members with no active email address?

**Review by nanderv — COMMENTED** [review]
looks good, just two comments.

File: datamining/views.py (hunk 1 above)
Comment (nanderv) [inline datamining/views.py:69]: Doesn't member.get_email just return a singular email address?
I'd expect something akin to:
```
for member in found_members:
  email = member.get_email()
  if email is not None:
    member_mails.append(email)
```

File: members/models/member.py (hunk 2 above)
Comment (nanderv) [inline members/models/member.py:70]: Why send emails to members with no active email address?

**Review by nanderv — CHANGES_REQUESTED** [review]
Looks good, just some small comments

File: datamining/views.py (hunk 1 above)
Comment (RemcoSwenker) [reply datamining/views.py:69]: I'm not sure but that is what you wrote 2 months ago with the update of the member filter. I would have to check for None and Blank I would guess.

File: members/models/member.py (hunk 2 above)
Comment (RemcoSwenker) [reply members/models/member.py]: good point i misread the issue about that so i fixed it

File: datamining/views.py (hunk 1 above)
Comment (nanderv) [reply datamining/views.py:69]: oh lol, it's python so it checks for the length of the string xD

Comment (nanderv) [reply datamining/views.py:69]: Perhaps a simple `if email:` is correct here.

Comment (RemcoSwenker) [reply datamining/views.py:70]: I did together with none

File: datamining/views.py [hunk 3]
```diff
@@ -42,8 +42,9 @@ def show_members_by_date(request):
     today = datetime.date.today().strftime("%Y-%m-%d")
 
     for member in found_members:
-        if len(member.get_email()) > 0:
-            member_mails.append(member.get_email())
+        email = member.get_email()
+        if email is not None and email:
```
Comment (nanderv) [inline datamining/views.py:46]: A bit pedantic, but None values are falsy anyway, so this is the same as 
`if email`

Comment (RemcoSwenker) [reply datamining/views.py]: fix

**Review by nanderv — APPROVED** [review]

**Pull request merged**